### PR TITLE
Update ERC721 example

### PR DESCRIPTION
**Pull request by ssteiger**
### - What I did

Added a comment & moved function call after 'assert' to save gas

### - Cute Animal Picture

image

### CODE REVIEW

**Comment by fubuloubu** [comment]
So, I'll have to think about this one a little bit more, but this pattern of making state changes prior to external calls is a re-entrancy protection mechanism. Not sure if that entirely holds here, but I definitely think that needs to be analyzed and explained in a comment if true.

File: examples/tokens/ERC721.vy [hunk 1]
```diff
@@ -272,10 +272,11 @@ def safeTransferFrom(
     @param _tokenId The NFT to transfer.
     @param _data Additional data with no specified format, sent in call to `_to`.
     """
-    self._transferFrom(_from, _to, _tokenId, msg.sender)
     if _to.is_contract: # check if `_to` is a contract address
         returnValue: bytes32 = ERC721Receiver(_to).onERC721Received(msg.sender, _from, _tokenId, _data)
+        # Throws if transfer destination is a contract which does not implement 'onERC721Received'
         assert returnValue == method_id("onERC721Received(address,address,uint256,bytes)", bytes32)
+    self._transferFrom(_from, _to, _tokenId, msg.sender)
```
Comment (fubuloubu) [inline examples/tokens/ERC721.vy]:
```suggestion
```

File: examples/tokens/ERC721.vy [hunk 2]
```diff
@@ -272,10 +272,11 @@ def safeTransferFrom(
     @param _tokenId The NFT to transfer.
     @param _data Additional data with no specified format, sent in call to `_to`.
     """
-    self._transferFrom(_from, _to, _tokenId, msg.sender)
```
Comment (fubuloubu) [inline examples/tokens/ERC721.vy]: Add this line back

**Comment by jacqueswww** [comment]
@ssteiger please see comments above.

**Comment by ssteiger** [comment]
I created a new pull request #1279.
I'll delete this branch once #1279 is merged.

**Comment by fubuloubu** [comment]
Closing in favor of #1279

**Pull request closed without merging**